### PR TITLE
feat: [CDS-72725]: Implementing CustomObject to get and set field from object

### PR DESCRIPTION
**Pull request by GoelPratyush99**
## Describe your changes
Implement Customs Object and CustomObjectFactory to read and update the KubernetesResource Object (like Virtual Services and DestinationRules) to remove dependency from fabric8

## Checklist
- [x] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51185)

### CODE REVIEW

**Comment by GoelPratyush99** [comment]
trigger uts

**Comment by GoelPratyush99** [comment]
trigger ut2

**Comment by GoelPratyush99** [comment]
trigger ut0

**Comment by GoelPratyush99** [comment]
trigger ut2

**Review by TarunUba1 — APPROVED** [review]

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/customObjects/ListCustomObject.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import io.harness.annotations.dev.HarnessModuleComponent;
+import io.harness.annotations.dev.ProductModule;
+
+import com.google.inject.Inject;
+import java.util.List;
+
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_K8S})
+public class ListCustomObject implements CustomObject {
+  private Object value;
+  @Inject CustomObjectFactory customObjectFactory;
```
Comment (abosii-harness) [inline 970-api-services-beans/src/main/java/io/harness/k8s/model/customObjects/ListCustomObject.java:22]: How exactly this is injected in object?

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/customObjects/ListCustomObject.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    this.value = object;
+  }
+
+  public List<Object> getObject() {
+    return (List<Object>) this.value;
+  }
+
+  @Override
+  public CustomObject getNode(String fieldName) {
+    List<Object> listNode = getObject();
```
Comment (abosii-harness) [inline 970-api-services-beans/src/main/java/io/harness/k8s/model/customObjects/ListCustomObject.java:34]: What about the use case list[n]?

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/customObjects/MapCustomObject.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    return (Map<String, Object>) this.value;
+  }
+
+  @Override
+  public CustomObject getNode(String fieldName) {
+    Map<String, Object> mapNode = getObject();
+    if (isEmpty(mapNode)) {
+      return null;
+    }
+    if (mapNode.containsKey(fieldName)) {
```
Comment (abosii-harness) [inline 970-api-services-beans/src/main/java/io/harness/k8s/model/customObjects/MapCustomObject.java:37]: What if field name is key.anotherKey?

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/customObjects/MapCustomObject.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      return customObjectFactory.create(mapNode.get(fieldName));
+    }
+    for (Object node : mapNode.values()) {
+      CustomObject customObject = customObjectFactory.create(node).getNode(fieldName);
```
Comment (abosii-harness) [inline 970-api-services-beans/src/main/java/io/harness/k8s/model/customObjects/MapCustomObject.java:41]: I'm not getting overall logic, how you're folding the fieldName?

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/customObjects/PrimitiveCustomObject.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_K8S})
+public class PrimitiveCustomObject implements CustomObject {
+  private Object value;
+
+  @Override
+  public CustomObject getNode(String fieldName) {
```
Comment (abosii-harness) [inline 970-api-services-beans/src/main/java/io/harness/k8s/model/customObjects/PrimitiveCustomObject.java:22]: Something is missing here, you can't just return null

**Review by abosii-harness — CHANGES_REQUESTED** [review]

**Pull request closed without merging**